### PR TITLE
Added support for passing Cargo.toml file path

### PR DESCRIPTION
**Pull request by daubaris**
Regarding issue #31. I am not sure if additional tests are needed since errors are handled by `cargo_toml` crate and forwarded to `cargo_scout`, but I can add them if needed.

### CODE REVIEW

**Comment by o0Ignition0o** [comment]
This is a great contribution, thanks a lot !
I think you're correct, the errors are handled and forwarded so it's probably fine :)

A couple of nits/options to eventually consider, and we're good to merge !

File: src/main.rs [hunk 1]
```diff
@@ -48,7 +52,10 @@ fn main() -> Result<(), error::Error> {
         .set_verbose(opts.verbose)
         .get_sections(&opts.branch)?;
     println!("Checking Cargo manifest");
-    let manifest = cargo::Parser::from_manifest_path("./Cargo.toml")?;
+    let path = opts
+        .cargo_toml
+        .unwrap_or_else(|| String::from("./Cargo.toml"));
```
Comment (o0Ignition0o) [inline src/main.rs]: Are we looking for [unwrap_or](https://doc.rust-lang.org/std/option/enum.Option.html#method.unwrap_or) here ?

File: src/main.rs [hunk 2]
```diff
@@ -28,6 +28,10 @@ struct Options {
     )]
     /// Set the target branch
     branch: String,
+
+    #[structopt(short = "t", long = "cargo-toml")]
```
Comment (o0Ignition0o) [inline src/main.rs]: [structopt seems to allow to pass a default value](https://docs.rs/structopt/0.3.5/structopt/?search=#how-to-derivestructopt)

Maybe the cargo_toml path could be a String, thanks to this feature?

**Review by o0Ignition0o — COMMENTED** [review]
This is a very good contribution 🎉 a couple of nits and we're good to go ! :)

File: src/main.rs (hunk 1 above)
Comment (daubaris) [reply src/main.rs]: Yeah, initially I wanted to go with `unwrap_or`, but `clippy_pedantic` suggests `unwrap_or_else`

File: src/main.rs (hunk 2 above)
Comment (daubaris) [reply src/main.rs]: That's true. It makes more sense to have it as a `default_value` rather than an `Option<String>`

File: src/main.rs (hunk 1 above)
Comment (o0Ignition0o) [reply src/main.rs]: Oh that's funny, I gotta give it a try on my computer, I wonder why clippy says that Oo

Comment (o0Ignition0o) [reply src/main.rs]: https://rust-lang.github.io/rust-clippy/master/index.html#or_fun_call

"The function will always be called and potentially allocate an object acting as the default."

Oh I didn't know that, TIL ! thank you for pointing me to it ! :D

**Comment by o0Ignition0o** [comment]
It looks great ! I'm going to merge it.
Thank's a lot for your contribution! 🎉

**Pull request merged**